datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeDrStatusResponseConverter.java

```diff
@@ -0,0 +1,24 @@
+package com.sequenceiq.cloudbreak.datalakedr.converter;
+
+import com.cloudera.thunderhead.service.datalakedr.datalakeDRProto;
+import com.sequenceiq.cloudbreak.datalakedr.model.DatalakeDrStatusResponse;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GrpcStatusResponseToDatalakeDrStatusResponseConverter {
+
+    public DatalakeDrStatusResponse convert(datalakeDRProto.BackupDatalakeStatusResponse response) {
+        return new DatalakeDrStatusResponse(
+            DatalakeDrStatusResponse.State.valueOf(response.getOverallState()),
+            Optional.ofNullable(response.getFailureReason())
+        );
```

Comment:
we can not depend on just the getOverallState. Here you need to check the internal states as well.

If the overall status could be failed but the internal states could be still in progress.

Reply:
My thought there was that if the overall state was failed, but some operations were still in progress, that we no longer had to block the flow. At that point it doesn't matter if something interrupts the b/r operation, because even if the remaining operations succeed, we still end up in a bad state. I'm pretty confident that's true for backup flows. However, I'm not as sure about restore flows. If part of a restore has failed, but there are ongoing operations, would we wind up in a worse position if we interrupted those operations? Or would it not matter because we would have to re-attempt the restore regardless?

Reply:
I agree with you with the internal behavior. Why do we want to report that backup-restore operations are complete even when some of the internal operations are still in progress when it can be reported accurately with a simple check?

Reply:
It's not a question of reporting, it's a question of blocking other flows. If the Solr backup has failed, but the HBase backup is still ongoing, do we want to block other datalake flows until the HBase backup is complete? Or do we want to say that the overall backup has failed, so don't bother waiting for the HBase part to finish and go ahead and unblock the other flows?

Reply:
I'm resolving this conversation because it is likely going to be rendered a moot point but an upcoming change in the datalake-dr service.